go/mysql/mysql56_gtid_set.go

```diff
@@ -334,6 +334,72 @@ func (set Mysql56GTIDSet) AddGTID(gtid GTID) GTIDSet {
 	return newSet
 }
 
+// Union implements GTIDSet.Union().
+func (set Mysql56GTIDSet) Union(other GTIDSet) GTIDSet {
+	if set == nil || other == nil {
```

Comment:
same comment as for MariadbGTIDSet - shouldn't we keep the non-nil set?

Reply:
I think I'm now starting to understand your point. Are you saying that in the case that `set` is nil, but `other` is not, we should return `other`?

Reply:
Bingo

Reply:
I'm on the fence about this. Should calling a method with a nil receiver be handled as a "valid" call? In the case of returning the non-nil set, that's what we're doing. If we are ok with considering nil receiver method calls valid, then I would agree that the union between a nil and non-nil set, should return the non-nil set. Thoughts?

Reply:
I just realized the main discussion is now going on above at the mariadb comment, so I'll remove this ^ from here and post it up there.

Reply:
+1

Reply:
Per the discussion above and the consensus we came to, this has been fixed in line with your suggestion: https://github.com/vitessio/vitess/pull/6180/commits/aab2a6547eddcef8318d715433785bb502c3c295.